src/setuptools_scm/_integration/pyproject_reading.py

```diff
@@ -224,6 +228,18 @@ def read_pyproject(
         requires,
     )
 
+    setuptools_dynamic_version = (
+        defn.get("tool", {})
+        .get("setuptools", {})
+        .get("dynamic", {})
+        .get("version", None)
+    )
+    if setuptools_dynamic_version is not None:
+        warnings.warn(
+            f"{path}: at [tool.setuptools.dynamic]\n"
+            "version = {attr = ...} is sabotaging setuptools-scm"
```

Comment:
I understand you don't appreciate this hack that forces you to make another release just to deprecate it, but I read the warning as something that implies intent, could it be reworded?

It would also be nice if the warning contained something actionable for the user, right now the user knows they're sabotaging setuptools-scm but don't get any indication of what they should do instead (or where to read about alternative solutions).

Reply:
@pjonsson id appreciate feedback on the updated warning

Reply:
@RonnyPfannschmidt sorry, missed this. Besides the typos I commented on in the other message, I think the updated message is much better, it gives me as a Python beginner a reasonable idea of what the expectations are and what I should do to fix the problem.